src/UrlBuilder.ts

```diff
@@ -0,0 +1,28 @@
+import { UrlBuilderType } from './types/type';
+
+export const UrlBuilder:UrlBuilderType = {
+  config: null,
+
+  init(config) {
+    this.config = { ...config, scopes: JSON.stringify(config.scopes) };
+    return {
+      createSignUpUrl: this.createSignUpUrl.bind(this),
+      createSignInUrl: this.createSignInUrl.bind(this),
+      createLogoutUrl: this.createLogoutUrl.bind(this),
+    };
+  },
+  createSignUpUrl() {
+    return `${this.config?.issuer}/web/register`
+      + `?appId=${this.config?.appId}&scopes=${this.config?.scopes}&callbackUrl=${this.config?.redirectUri}`;
+  },
+
+  createSignInUrl() {
+    return `${this.config?.issuer}/web/login`
+      + `?appId=${this.config?.appId}&scopes=${this.config?.scopes}&callbackUrl=${this.config?.redirectUri}/callback`;
```

Comment:
I think you don't need the trailing `/callback` as that (if needed) is a part of redirectUri itself.